### DISCUSSION TITLE
Stay consistently on the same side of the road (pedestrian routing?)

### DISCUSSION THREAD

**Discussion by dschwen** [valhalla/valhalla#3811] [Q&A]
How can I get a route that doesn't require crossing the road multiple times between successive stops (e.g. with multiple stops along the same street). I don;t necessarily want to restrict to pedestrian routing only, because the route may contain car only portions. Think two people, one driving a car, and the second one jumping out and delivering something to a door. Crossing the road would be inconvenient, potentially dangerous, and would add additional zig-zag distance.

**Comment by kevinkreiser** [comment]
we have a location type called `through` which requires the route to continue in the direction it was traveling rather than turning around. we also have a maneuver penalty should should reduce the number of the maneuvers in the route generally. would you mind sending a sample request or an image of a map with a route line drawn on it to illustrate your issue. im not sure if ive actually understood what youre trying to avoid. thanks!

EDIT: maybe you are talking about a dual carriage way with stops on both sides of the dual carriage way where you are fine visiting the other side of the street by parking on only eg the south bound side? when i say dual carriage way i mean like a road divided by a median so that in the map its "digitized" as two separate roads, though the road itself has the same name and to humans is semantically the same road. is this what you are talking about?
